modules/class_optimize.py

```diff
@@ -1,307 +1,233 @@
-from modules.class_ems import EnergieManagementSystem, Wechselrichter
+from modules.class_ems import EnergieManagementSystem
 from modules.class_haushaltsgeraet import Haushaltsgeraet
+from modules.class_inverter import Wechselrichter
 from modules.visualize import visualisiere_ergebnisse
 
-matplotlib.use("Agg")  # Setzt das Backend auf Agg
-import random
-from datetime import datetime, timedelta
-
-from deap import algorithms, base, creator, tools
-
 sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 from config import moegliche_ladestroeme_in_prozent
 
 
-def isfloat(num):
+def isfloat(num: Any) -> bool:
+    """Check if a given input can be converted to float."""
     try:
         float(num)
         return True
     except ValueError:
         return False
 
 
-def differential_evolution(
-    population,
-    toolbox,
-    cxpb,
-    mutpb,
-    ngen,
-    stats=None,
-    halloffame=None,
-    verbose=__debug__,
-):
-    """Differential Evolution Algorithm"""
-
-    # Evaluate the entire population
-    fitnesses = list(map(toolbox.evaluate, population))
-    for ind, fit in zip(population, fitnesses):
-        ind.fitness.values = fit
-
-    if halloffame is not None:
-        halloffame.update(population)
-
-    logbook = tools.Logbook()
-    logbook.header = ["gen", "nevals"] + (stats.fields if stats else [])
-
-    for gen in range(ngen):
-        # Generate the next generation by mutation and recombination
-        for i, target in enumerate(population):
-            a, b, c = random.sample([ind for ind in population if ind != target], 3)
-            mutant = toolbox.clone(a)
-            for k in range(len(mutant)):
-                mutant[k] = c[k] + mutpb * (a[k] - b[k])  # Mutation step
-                if random.random() < cxpb:  # Recombination step
-                    mutant[k] = target[k]
-
-            # Evaluate the mutant
-            mutant.fitness.values = toolbox.evaluate(mutant)
-
-            # Replace if mutant is better
-            if mutant.fitness > target.fitness:
-                population[i] = mutant
-
-        # Update hall of fame
-        if halloffame is not None:
-            halloffame.update(population)
-
-        # Gather all the fitnesses in one list and print the stats
-        record = stats.compile(population) if stats else {}
-        logbook.record(gen=gen, nevals=len(population), **record)
-        if verbose:
-            print(logbook.stream)
-
-    return population, logbook
-
-
 class optimization_problem:
-    def __init__(self, prediction_hours=24, strafe=10, optimization_hours=24):
-        self.prediction_hours = prediction_hours  #
+    def __init__(
+        self,
+        prediction_hours: int = 24,
+        strafe: float = 10,
+        optimization_hours: int = 24,
+        verbose: bool = False,
+        fixed_seed: Optional[int] = None,
+    ):
+        """Initialize the optimization problem with the required parameters."""
```

Comment:
Docstring can be more helpful to describe each parameter. I'm using [Google Style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings), maybe we can discuss on using in this project.

Reply:
Lets discuss this in the discussion section.